lib/parallel_cucumber_worker/cli/main.js

```diff
@@ -37,67 +37,17 @@ var Main = function() {
     }
 
     self.Cucumber = require(cucumberPath);
-    var RealAstTreeWalker = self.Cucumber.Runtime.AstTreeWalker;
 
-    self.Cucumber.Runtime.AstTreeWalker = function(features, supportCodeLibrary, listeners, options) {
-      var self2 = RealAstTreeWalker(features, supportCodeLibrary, listeners, options);
 
-      var realBroadcastEvent = self2.broadcastEvent;
-      var realGetListeners = supportCodeLibrary.getListeners;
-
-      self2.broadcastEvent = function(event, callback) {
-        if (isCensoredEvent(event)) {
-          disableSupportCodeListeners();
-          realBroadcastEvent(event, function() {
-            enableSupportCodeListeners();
-            callback();
-          });
-        }
-        else {
-          realBroadcastEvent(event, callback);
-        }
-      };
-
-      self2.broadcastEventUncensored = function(event, callback) {
-        realBroadcastEvent(event, callback);
-      };
-
-      function disableSupportCodeListeners() {
-        supportCodeLibrary.getListeners = function() {
-          return [];
-        };
-      }
-
-      function enableSupportCodeListeners() {
-        supportCodeLibrary.getListeners = realGetListeners;
-      }
-
-      function isCensoredEvent(event) {
-        var name = event.getName();
-
-        if (name === 'BeforeFeatures') {
-          if (self.beforeFeatures) {
-            self.beforeFeatures = false;
-            return false;
-          }
-
-          return true;
-        }
-        else if (name === 'AfterFeatures') {
-          return true;
-        }
-        else {
-          return false;
-        }
-      }
-
-      self.astTreeWalker = self2;
-      return self2;
+    self.Cucumber.Runtime.EventBroadcaster = function(listeners, supportCodeLibrary) {
+      var RealEventBroadcaster = require('../../parallel_cucumber/event_broadcaster.js');
+      var eventBroadCasterInstance = RealEventBroadcaster(listeners, supportCodeLibrary);
```

Comment:
@ViacheslauBushyla how about assigning the object directly to the field, without declaring the variable?

Reply:
@ViacheslauBushyla I notice you return the variable so you can ignore my comment above.